Azure/sonic-buildimage
# [device/celestica]: Add firmware management api based on the new platform API

**- What I did**
- Add firmware management function base (https://github.com/Azure/sonic-platform-common/pull/34)
- Implement part of the Chassis, Module and Device related APIs.
- Chassis API
  - get_firmware_version(component_name)
  - install_component_firmware(component_name, image_path)
- Component API
  - get_firmware_version()
  - upgrade_firmware(image_path)
- Device API
  - get_name()
  - get_name_list()

**- How I did it**
1. Implement APIs based on sonic platform API base
2. Add Component API to support chassis component firmware management

**- How to verify it**
- Install CPLD upgrade tool (https://github.com/Azure/sonic-buildimage/pull/2446)
- Create python script follow the new platform API design doc to verify it (Azure/SONiC#285.)

Signed-off-by: Wirut Getbamrung wgetbumr@celestica.com
